lib/logger.c

```diff
@@ -354,40 +420,40 @@ void
 grn_default_logger_set_path(const char *path)
 {
   if (logger_inited) {
```

Comment:
What referring below is out of scope this PR.

I guess it potentially has a problem if  `grn_default_logger_set_path` and `grn_logger_init` and `grn_logger_fin`  are called concurrently because `logger_inited` can be replaced during execution of `grn_default_logger_set_path`.
However, it seems that these functions are never called concurrently and this problem never surface...

Reply:
It may be occurred but we don't support the situation. Initialization must be done from single thread.